src/components/map/MapComponent.tsx

```diff
@@ -18,7 +18,18 @@ const MapComponent: React.FC<MapProps> = ({ items }) => {
                 url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
             />
             {items.map(item => (
-                <Pin {...item} key={item.id} />
+                <Pin
+                    key={item.id}
+                    id={item.id}
+                    title={item.title}
+                    img={'img' in item ? item.img : item.images[0]}  // Handle img or images
+                    bedroom={'bedroom' in item ? item.bedroom : item.bedRooms}
+                    bathroom={item.bathroom}
+                    price={item.price}
+                    address={item.address}
+                    latitude={item.latitude}
+                    longitude={item.longitude}
+                />
```

Comment:
Destruct `item` instead of use `item`with dot. 
Or u can even use spread operator like
```
  <Pin
    key={item.id}
    {...item}
    img={'img' in item ? item.img : item.images[0]}  // Handle img or images
    bedroom={'bedroom' in item ? item.bedroom : item.bedRooms}
    />
```